lib/rubocop/cop/style/commented_keyword.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for comments put on the same line as some keywords.
+      # These keywords are: `begin`, `class`, `def`, `end`, `module`.
+      #
+      # Note that some comments (such as `:nodoc:` and `rubocop:disable`) are
+      # allowed.
+      #
+      # @example
+      #   # bad
+      #   if condition
+      #     statement
+      #   end # end if
+      #
+      #   # bad
+      #   class X # comment
+      #     statement
+      #   end
+      #
+      #   # bad
+      #   def x; end # comment
+      #
+      #   # good
+      #   if condition
+      #     statement
+      #   end
+      #
+      #   # good
+      #   class x # :nodoc:
+      #     y
+      #   end
+      class CommentedKeyword < Cop
+        MSG = 'Do not add comments on the same line as the ' \
+              '`%s` keyword.'.freeze
+
+        def investigate(processed_source)
+          heredoc_lines = extract_heredoc_lines(processed_source.ast)
+
+          processed_source.lines.each_with_index do |line, index|
+            next if heredoc_lines.any? { |r| r.include?(index + 1) }
+            next unless offensive?(line)
+
+            range = source_range(processed_source.buffer,
+                                 index + 1,
+                                 (line.index('#'))...(line.length))
+
+            add_offense(range, range)
+          end
+        end
+
+        private
+
+        def keywords
+          %w[begin class def end module]
```

Comment:
This should be a constant, not a method.

Reply:
@Drenmi @pocke More useful cop ideas. I wonder how we didn't create this one. I've seen such inline constants many times.